cumulus/polkadot-parachain/src/service.rs

```diff
@@ -972,11 +975,15 @@ pub async fn start_rococo_parachain_node(
 				client.clone(),
 			);
 
-			let params = BasicAuraParams {
+			let params = AuraParams {
 				create_inherent_data_providers: move |_, ()| async move { Ok(()) },
 				block_import,
-				para_client: client,
+				para_client: client.clone(),
+				para_backend: backend.clone(),
 				relay_client: relay_chain_interface,
+				code_hash_provider: move |block_hash| {
+					client.code_at(block_hash).ok().map(ValidationCode).map(|c| c.hash())
```

Comment:
Can't this be just one map?

Reply:
Done in d2e76ff6aef3efaa0d3ce3bdb696f11b379615e7.